internal/providers/docker/compose/servicedependency.go

```diff
@@ -54,16 +52,15 @@ func (sd *ServiceDependencies) UnmarshalYAML(unmarshal func(interface{}) error)
 		return nil
 	}
 
-	var asMap yaml.MapSlice
+	var asMap map[string]interface{}
```

Comment:
This introduces various bugs in the form of instability. Go maps are intentionally unordered keys. If you're serializing to a map, the encoder might sort the keys, but that's not the right thing here. We want to preserve the key order when we convert to a slice below (and in several other places throughout the code where you changed this).

Reply:
Why? I was treating these arrays as sets rather than ordered lists. Is the order pertinent for anything other than serialisation? My intent was to sort them at the point of serialisation to preserve the order of the original map.